StreamChatClient_v3/APIClient/RequestEncoder.swift

```diff
@@ -191,8 +201,9 @@ private extension URL {
     }
 }
 
-protocol ConnectionIdProviderDelegate: AnyObject {
+protocol ConnectionDetailsProviderDelegate: AnyObject {
     func provideConnectionId(completion: @escaping (_ connectionId: ConnectionId?) -> Void)
+    func provideToken() -> Token?
```

Comment:
Why isn't this async? 🤯

Reply:
In the current setup, it doesn't need to be. Maybe it will have to be async once we implement the token refresh functionality, but I didn't want to complicate things now.